src/scripts/train/OLMo3-1B.py

```diff
@@ -0,0 +1,172 @@
+"""
+Train a 1B OLMo model. Run this script without any arguments to see usage info.
+"""
+from datetime import datetime
+
+from olmo_core.config import DType
+from olmo_core.distributed.parallel import DataParallelType
+from olmo_core.internal.common import CLUSTER_TO_GPU_TYPE
+from olmo_core.internal.experiment import CommonComponents, main
+from olmo_core.nn.attention import SlidingWindowAttentionConfig
+from olmo_core.nn.transformer import TransformerConfig
+from olmo_core.optim import OptimGroupOverride, SchedulerUnits, SkipStepAdamWConfig
+from olmo_core.optim.scheduler import WSD
+from olmo_core.train import Duration, TrainerConfig
+from olmo_core.train.callbacks import (
+    BatchSizeSchedulerCallback,
+    CheckpointerCallback,
+    CometCallback,
+    WandBCallback,
+)
+from olmo_core.train.train_module import (
+    TransformerDataParallelConfig,
+    TransformerDataParallelWrappingStrategy,
+    TransformerTrainModuleConfig,
+)
+
+SEQUENCE_LENGTH = 8192
+GLOBAL_BATCH_SIZE = (
+    1024 * 4096
+)  # batch size at step 0, let's keep this independent of the sequence length in case we change it.
+MAX_DURATION = int(
+    10e12
+)  # Setting this higher than 6T (expected run time), in case we get to run longer since 1) we're using WSD and 2) our anneal will use different data
+ANNEAL_TOKENS = int(100e9)
+LR = (
+    4.4e-5 * 2
+)  # Based on 6T tokens with 100B anneal, don't forget to adjust when max duration or anneal length changes.
+SAVE_INTERVAL = 10000
+EVAL_INTERVAL = 1000
+
+
+def build_model_config(common: CommonComponents) -> TransformerConfig:
+    config = TransformerConfig.olmo2_1B(
+        vocab_size=common.tokenizer.padded_vocab_size(),
+        n_layers=16,
+        hidden_size_multiple_of=1024,
+    )
+    #  config.block.name = TransformerBlockType.default
+    #  config.block.attention.qk_norm = None
+    config.block.attention.sliding_window = SlidingWindowAttentionConfig(
+        force_first=False, pattern=[False, False, False, True]
+    )
+    config.block.attention.use_flash = True
+    config.block.attention.use_head_qk_norm = True
+
+    return config
+
+
+def build_train_module_config(common: CommonComponents) -> TransformerTrainModuleConfig:
+    rank_microbatch_size = 4 * SEQUENCE_LENGTH
+    if common.launch is not None:
+        gpus = {CLUSTER_TO_GPU_TYPE.get(c, "unknown") for c in common.launch.clusters}
+        if all("B200" in g for g in gpus):
+            rank_microbatch_size *= 2
+
+    return TransformerTrainModuleConfig(
+        rank_microbatch_size=rank_microbatch_size,
+        max_sequence_length=common.dataset.effective_sequence_length,
+        optim=SkipStepAdamWConfig(
```

Comment:
To maintain reproducibility of Olmo3 models need to flip `step_increment_bugfix` to False

Reply:
Done: 688fadca, also changed in the 7B script